internal/watch/notify_test.go

```diff
@@ -338,13 +338,14 @@ func TestWatchNonexistentDirectory(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	file := f.JoinPath("root", "parent", "a")
+	parent := f.JoinPath("parent")
+	file := f.JoinPath("parent", "a")
 
-	f.watch(file)
+	f.watch(parent)
```

Comment:
i thought this test was particularly for the case where someone is watching a Dockerfile that doesn't exist yet. I don't think this change preserves that meaning?

Reply:
Whoops, missed that! Good catch @nicks

Reply:
I've now fixed the test and added a test that doesn't work (which matches the original test).

Watching a file within a directory where neither exist is racey on Linux (before this change!), and fixing it would be involved.